### ISSUE TITLE
Use SequenceEqual() in String.Replace()

### ISSUE THREAD

**Issue by iSazonov** [dotnet/corefx#33005]
As metioned in #16214 we should use [SequenceEqual](https://source.dot.net/#System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs,864) in [String.Replace()](https://source.dot.net/#System.Private.CoreLib/shared/System/String.Manipulation.cs,1083)

**Comment by stephentoub** [comment]
> we should use 

We should measure it and see if it generally improves things, and if it does, use it.

**Comment by iSazonov** [comment]
Yes, I saw that SequenceEqual() uses SIMD acceleration so it made sense to open the tracking issue.
